### PR TITLE
[Bug](status) process error status on es_scroll_parser and compaction_action

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
process error status on es_scroll_parser and compaction_action

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/common/status.h [hunk 1]
```diff
@@ -501,6 +502,8 @@ class [[nodiscard]] Status {
 
     friend std::ostream& operator<<(std::ostream& ostr, const Status& status);
 
+    std::string msg() const { return _err_msg ? _err_msg->_msg : ""; }
```
Comment (yiguolei) [inline be/src/common/status.h:505]: rename to  error_msg()

File: be/src/olap/tablet.cpp [hunk 2]
```diff
@@ -1785,7 +1785,7 @@ Status Tablet::prepare_compaction_and_calculate_permits(CompactionType compactio
         if (!res.ok()) {
             set_last_base_compaction_failure_time(UnixMillis());
             *permits = 0;
-            if (!res.is<BE_NO_SUITABLE_VERSION>()) {
+            if (!res.is<BE_NO_SUITABLE_VERSION>() && !res.is<FULL_NO_SUITABLE_VERSION>()) {
```
Comment (zhannngchen) [inline be/src/olap/tablet.cpp:1788]: don't change here, this is error code for base compaction, and `FULL_NO_SUITABLE_VERSION` is generated by full compaction.

File: be/src/olap/tablet.cpp [hunk 3]
```diff
@@ -1805,7 +1805,7 @@ Status Tablet::prepare_compaction_and_calculate_permits(CompactionType compactio
         if (!res.ok()) {
             set_last_full_compaction_failure_time(UnixMillis());
             *permits = 0;
-            if (!res.is<BE_NO_SUITABLE_VERSION>()) {
+            if (!res.is<BE_NO_SUITABLE_VERSION>() && !res.is<FULL_NO_SUITABLE_VERSION>()) {
```
Comment (zhannngchen) [inline be/src/olap/tablet.cpp:1808]:
```suggestion
            if (!res.is<FULL_NO_SUITABLE_VERSION>()) {
```

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by zhangstar333 — APPROVED** [review]

**Comment by BiteTheDDDDt** [comment]
do not pick this to 2.0

**Pull request merged**